src/index.js

```diff
@@ -46,28 +46,46 @@ function hasParams(required, actual) {
 }
 
 /**
- * Create providers along index definitions and return an array of objects
- * consisting of an index definition and its providers
+ * Create handlers along index definitions and return an array of objects
+ * consisting of an index definition and its handler.
  *
- * @param {Array} indices array of index definitions
- * @returns array of index definitions and their providers
+ * @param {Array} configs array of index definitions
+ * @param {object} params parameters available
+ * @param {object} log logger
+ *
+ * @returns array of index definitions and their handlers
  */
-function createProviders(indices, params, log) {
-  return indices
-    .map((config) => {
-      let provider;
-      const candidate = providers
-        .filter((c) => hasParams(c.required, params))
-        .find((c) => c.match(config.target));
-      if (candidate) {
-        try {
-          provider = candidate.create(params, config, log);
-        } catch (e) {
-          log.error(`Unable to create provider for ${config.name}`, e);
-        }
+function createHandlers(configs, params, log) {
```

Comment:
Instead of creating one provider per index, let provider decide whether it wants to multiple providers or a composite provider (as is the case in Excel)